diff --git a/convert.py b/convert.py
index ac254c3..2ae11f6 100644
--- a/convert.py
+++ b/convert.py
@@ -28,7 +28,10 @@ for subset in ['val', 'test', 'train']:
     category_sourcefile = 'class-descriptions-boxable.csv'
     if subset == 'train':
         image_sourcefile = 'train-images-boxable-with-rotation.csv'
-        annotation_sourcefile = 'train-annotations-bbox.csv'
+        if args.version == 6:
+          annotation_sourcefile = 'oidv6-train-annotations-bbox.csv'
+        else:
+          annotation_sourcefile = 'train-annotations-bbox.csv'
         image_label_sourcefile = 'train-annotations-human-imagelabels-boxable.csv'
     elif subset == 'val':
         image_sourcefile = 'validation-images-with-rotation.csv'
diff --git a/utils.py b/utils.py
index 7ec24c7..fbdb198 100644
--- a/utils.py
+++ b/utils.py
@@ -78,7 +78,11 @@ def convert_image_annotations(original_image_metadata, original_image_annotation
             img['license'] = licenses_by_url_http[license_url]['id']
 
         # Extract height and width
-        filename = os.path.join(image_dir, img['file_name'])
+        if not os.path.exists(image_dir):
+          filename = os.path.join(image_dir + "_"+img['file_name'][0].lower(), img['file_name'])
+        else:
+          filename = os.path.join(image_dir, img['file_name'])
+
         img['width'], img['height'] = imagesize.get(filename)
             
         # Add to list of images
